test/harry/main/org/apache/cassandra/harry/model/ASTSingleTableModel.java

```diff
@@ -772,6 +805,7 @@ private Pair<List<Clustering<ByteBuffer>>, List<Conditional>> splitOn(ImmutableU
         // pk requires equality
         Map<Symbol, List<ByteBuffer>> pks = new HashMap<>();
         List<Conditional> other = new ArrayList<>();
+        Map<Symbol, List<Conditional.Where>> rangeConditions = new HashMap<>();
```

Comment:
Why are you trying to handle ranges in this method?  we call this on partition and if the partition has clustering ranges we then need to special case it and not call `splitOn` for clustering.  I think we can drop the changes to this method